File: easy/replace_greatest_elements.py
def replace_greatest_elements(arr: list[int]) -> list[int]:
    max, right = -1, len(arr) - 1
    while right >= 0:
        temp = arr[right]
        arr[right] = max
        if temp > max:
            max = temp
        right -= 1
    return arr

File: easy/replace_greatest_elements.go
package easy

func ReplaceGreatestElements(arr []int) []int {
	max := -1
	if len(arr) == 0 {
		return arr
	}
	right := len(arr)
	for right > 0 {
		right -= 1
		temp := arr[right]
		arr[right] = max
		if temp > max {
			max = temp
		}
	}
	return arr
}
